File: 01-a.js
const fs = require('node:fs');
let answerA = 0;
try {
    const input = fs.readFileSync('01-input.txt', 'utf8').replaceAll("\r","");
    const lines = input.split("\n");
    lines.forEach((line) => {
        const digits = line.match(/\d/g);
        let lineScore = (10 * digits[0]) + (1 * digits[digits.length - 1]);
        //console.log(line, digits, lineScore);
        answerA += lineScore;
    });
}
catch(e) {
    console.error(e);
}

console.log("The answer to part 1 is:", answerA);

File: 2025/01-a.py
with open("2025/01-input.txt") as in_file:
    lines = in_file.read().strip().split("\n")

counter = 0
pointer = 50
for line in lines:
    direction = line[0:1]
    rotation = int(line[1:])
    if direction == "R":
        pointer = (pointer + rotation) % 100
    elif direction == "L":
        pointer = (pointer - rotation) % 100
    if pointer == 0:
        counter += 1

print(f"The answer is {counter}")